ShaderQuery : Add "From Selection" menu item

ericmehl opened the pull request:
This adds a menu item to the `shader` plug's popup menu showing a list of attributes attached to the selected locations which can be used for the shader attribute.

### Breaking changes ###

None

### Checklist ###

- [X] I have read the [contribution guidelines](https://github.com/GafferHQ/gaffer/blob/main/CONTRIBUTING.md).
- [X] I have updated the documentation, if applicable.
- [X] I have tested my change(s) in the test suite, and added new test cases where necessary.
- [X] My code follows the Gaffer project's prevailing coding style and conventions.

johnhaddon reviewed (changes requested):
LGTM, but I was hoping for this treatment on the `+` menu for adding _parameters_, rather than the menu for choosing _attributes_. It's a good feature for attributes too, but can we add to the parameters menu too before merging?

ericmehl commented:
> can we add to the parameters menu too before merging

I added that and squashed down into 9d95d488b95fc02851344acda9c5034ba8f975d9. CI is just about wrapped up successfully - ready for another look.

ericmehl commented:
I just did a quick rebase and push to resolve the `Changes.md` conflict, but otherwise it's unchanged.
